### DISCUSSION TITLE
Showing working changes not working on WSL Ubuntu on Windows 10

### DISCUSSION THREAD

**Discussion by roklenardic** [JanDeDobbeleer/oh-my-posh#1195] [General]
Hi, I've installed oh-my-posh on my windows 10 machine and installed it for Powershell and Ubuntu 20.04 WSL instance. However, the ubuntu instance does not show working changes whereas the powershell does. Both use the marcduiker theme.
![Screenshot_6](https://user-images.githubusercontent.com/121102/141016627-2d742264-0081-4c37-80e4-e68a95962b9e.png)
![Screenshot_7](https://user-images.githubusercontent.com/121102/141016678-fd415551-d525-4730-b6b6-af0940fc3d99.png)

Any idea?

**Comment by JanDeDobbeleer** [comment]
How did you install oh-my-posh? The reason I'm asking is that I want to understand if both versions are the same (or the theme file).

**Reply by roklenardic** [reply]
Hi Jan, 

the wsl/ubuntu version was installed using these instructions: https://ohmyposh.dev/docs/linux
and the powershell one was installed using the winget, again following instructions on https://ohmyposh.dev/docs/windows

Just did a diff between the two theme files and they are identical: 
![image](https://user-images.githubusercontent.com/121102/141081924-aca33eeb-c358-411a-aec0-871055a76904.png)

**Reply by JanDeDobbeleer** [reply]
And both versions are the same? When it comes to this setup, it's more interesting to install via [winget](https://ohmyposh.dev/docs/windows#installation). That will add both executables (windows and unix) to your system and results into 1 and the same version of oh-my-posh cross shell/env.

**Reply by roklenardic** [reply]
1. How could I check that both versions are the same?
2. Are you saying that what I installed within Powershell using winget could be used in the WSL Ubuntu bash as well? How?

**Reply by JanDeDobbeleer** [reply]
1. Powershell (if you used `Install-Module oh-my-posh`): `(Get-Module oh-my-posh).Version`, unix (and Windows if you used Winget): `oh-my-posh --version`
2. Yes, by using the following in bash: `eval "$(oh-my-posh-wsl --init --shell bash --config ~/.poshthemes/jandedobbeleer.omp.json)"`

**Reply by roklenardic** [reply]
Powershell 6.1.3
WSL Ubuntu 6.2.0

**Reply by JanDeDobbeleer** [reply]
Can you try to set it up with Winget alone for both shells? That way it's fully aligned (and point to the same config).

**Reply by roklenardic** [reply]
Even with the oh-my-posh-wsl I get no working changes, just one stash. And I do have one small change in README.md
![image](https://user-images.githubusercontent.com/121102/141111750-07338f4c-995f-4c0f-bd3e-99858d5b6973.png)

**Reply by JanDeDobbeleer** [reply]
@roklenardic can you check with the latest [version](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v6.2.1)? Beware it's not through on Winget, so keep an eye out. That one has more logging meaning you can do:

```bash
oh-my-posh-wsl --debug --config ~/.poshthemes/jandedobbeleer.omp.json --shell uni
```

This prints the timings and creates a log file at `~/oh-my-posh.log`. I would need the output from that file.

**Reply by roklenardic** [reply]
Is this ok?
![image](https://user-images.githubusercontent.com/121102/141249136-1d187ff6-1c36-4608-9197-1058004083bb.png)

My .bash rc init looks like this:
eval "$(/mnt/c/Users/rokle/scoop/apps/oh-my-posh/current/bin/oh-my-posh-wsl --debug --init --shell bash --config ~/.poshthemes/marcduiker.omp.json)"

the --shell uni was not accepted, gave me some error.

**Reply by JanDeDobbeleer** [reply]
@roklenardic you shouldn't add `--debug` to your eval script, you can run `oh-my-posh-wsl --debug --config ~/.poshthemes/jandedobbeleer.omp.json --shell uni` directly inside the path where you have the issue. That way we only have the relevant output.

**Reply by roklenardic** [reply]
Here we go :)
/mnt/c/Users/rokle/scoop/apps/oh-my-posh/current/bin/oh-my-posh-wsl --debug --config ~/.poshthemes/jandedobbeleer.omp.json --shell uni

Here are the timings of segments in your prompt:
ConsoleTitle(true)   -   1 ms - uni in FindAPI
session(true)        -   0 ms -  rok 
path(true)           -   0 ms -    FindAPI 
git(true)            - 435 ms -  dev ≢  1 
node(false)          -   1 ms -
go(false)            -   0 ms -
julia(false)         -   0 ms -
python(false)        -   0 ms -
ruby(false)          -   0 ms -
azfunc(false)        -   0 ms -
aws(false)           -   0 ms -
root(false)          -   0 ms -
executiontime(true)  -   0 ms -  羽0ms⠀
exit(true)           -   0 ms -   
shell(true)          -   0 ms -   uni 
ytm(false)           -   0 ms -
battery(false)       -   0 ms -
time(true)           -   0 ms -  10:32:50 

2021/11/11 10:32:49 #### start oh-my-posh run ####
2021/11/11 10:32:49 debug: getenv

2021/11/11 10:32:49 getenv duration: 38.6µs, args: XDG_CACHE_HOME
2021/11/11 10:32:49 getCachePath duration: 176.9µs, args:
2021/11/11 10:32:49 getArgs duration: 1.6µs, args:
2021/11/11 10:32:49 getArgs duration: 1.5µs, args:
2021/11/11 10:32:49 getShellName duration: 3.5µs, args:
2021/11/11 10:32:49 isRunningAsRoot duration: 10.5µs, args:
2021/11/11 10:32:49 getcwd duration: 232.6µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 18.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.8µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.4µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.4µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.4µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.4µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.4µs, args:
2021/11/11 10:32:49 getShellName duration: 2.8µs, args:
2021/11/11 10:32:49 getCurrentUser duration: 3.4µs, args:
2021/11/11 10:32:49 getHostName duration: 8.6µs, args:
2021/11/11 10:32:49 getcwd duration: 2.8µs, args:
2021/11/11 10:32:49 getCurrentUser duration: 2.9µs, args:
2021/11/11 10:32:49 getRuntimeGOOS duration: 7.4µs, args:
2021/11/11 10:32:49 debug: getenv

2021/11/11 10:32:49 getenv duration: 28.9µs, args: SSH_CONNECTION
2021/11/11 10:32:49 debug: getenv

2021/11/11 10:32:49 getenv duration: 32.4µs, args: SSH_CLIENT
2021/11/11 10:32:49 debug: getenv

2021/11/11 10:32:49 getenv duration: 26.1µs, args: POSH_SESSION_DEFAULT_USER
2021/11/11 10:32:49 getcwd duration: 2.8µs, args:
2021/11/11 10:32:49 getcwd duration: 2.4µs, args:
2021/11/11 10:32:49 getArgs duration: 2.6µs, args:
2021/11/11 10:32:49 getcwd duration: 2.5µs, args:
2021/11/11 10:32:49 getRuntimeGOOS duration: 2.5µs, args:
2021/11/11 10:32:49 getRuntimeGOOS duration: 2.4µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.4µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.4µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.4µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getPathSeperator duration: 2.5µs, args:
2021/11/11 10:32:49 getRuntimeGOOS duration: 2.5µs, args:
2021/11/11 10:32:49 getcwd duration: 2.6µs, args:
2021/11/11 10:32:49 getRuntimeGOOS duration: 2.4µs, args:
2021/11/11 10:32:49 debug: getenv
Ubuntu
2021/11/11 10:32:49 getenv duration: 28.1µs, args: WSL_DISTRO_NAME
2021/11/11 10:32:49 isWsl duration: 67.6µs, args:
2021/11/11 10:32:49 getcwd duration: 2.5µs, args:
2021/11/11 10:32:49 hasCommand duration: 6.9455ms, args: git.exe
2021/11/11 10:32:49 getcwd duration: 1.1µs, args:
2021/11/11 10:32:49 hasParentFilePath duration: 59.6µs, args: .git
2021/11/11 10:32:49 getRuntimeGOOS duration: 1.1µs, args:
2021/11/11 10:32:49 debug: getenv
Ubuntu
2021/11/11 10:32:49 getenv duration: 13.5µs, args: WSL_DISTRO_NAME
2021/11/11 10:32:49 isWsl duration: 28.6µs, args:
2021/11/11 10:32:49 getcwd duration: 1.1µs, args:
2021/11/11 10:32:50 error: runCommand
exit status 128
2021/11/11 10:32:50 runCommand duration: 425.7431ms, args: git.exe -C /mnt/c/code/somerepo/ --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --short --branch
2021/11/11 10:32:50 getFileContent duration: 259.2µs, args: /mnt/c/code/somerepo/.git/HEAD
2021/11/11 10:32:50 hasFolder duration: 95.4µs, args: /mnt/c/code/somerepo/.git/rebase-merge
2021/11/11 10:32:50 hasFolder duration: 86.2µs, args: /mnt/c/code/somerepo/.git/rebase-apply
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFilesInDir duration: 101.6µs, args: MERGE_MSG
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFilesInDir duration: 95.5µs, args: CHERRY_PICK_HEAD
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFilesInDir duration: 91.2µs, args: REVERT_HEAD
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFilesInDir duration: 93.6µs, args: sequencer/todo
2021/11/11 10:32:50 getFileContent duration: 267µs, args: /mnt/c/code/somerepo/.git/logs/refs/stash
2021/11/11 10:32:50 getcwd duration: 1.6µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 244.5µs, args: *.js
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.4µs, args:
2021/11/11 10:32:50 hasFiles duration: 191.9µs, args: *.ts
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 91.8µs, args: package.json
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 76.2µs, args: .nvmrc
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.4µs, args:
2021/11/11 10:32:50 hasFiles duration: 77.2µs, args: pnpm-workspace.yaml
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 83.6µs, args: .pnpmfile.cjs
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 75µs, args: .npmrc
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 202.3µs, args: *.go
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 116.4µs, args: go.mod
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.2µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 176.2µs, args: *.jl
2021/11/11 10:32:50 getcwd duration: 9.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.2µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 171.8µs, args: *.py
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 169µs, args: *.ipynb
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 96.7µs, args: pyproject.toml
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 85.2µs, args: venv.bak
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 119.2µs, args: venv
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.4µs, args:
2021/11/11 10:32:50 hasFiles duration: 73.9µs, args: .venv
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 208.9µs, args: *.rb
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 102.8µs, args: Rakefile
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 74.4µs, args: Gemfile
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 75µs, args: host.json
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 75µs, args: local.settings.json
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 getPathSeperator duration: 1.3µs, args:
2021/11/11 10:32:50 hasFiles duration: 84µs, args: function.json
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 debug: getenv

2021/11/11 10:32:50 getenv duration: 15.3µs, args: AWS_VAULT
2021/11/11 10:32:50 debug: getenv

2021/11/11 10:32:50 getenv duration: 14µs, args: AWS_PROFILE
2021/11/11 10:32:50 debug: getenv

2021/11/11 10:32:50 getenv duration: 13.8µs, args: AWS_REGION
2021/11/11 10:32:50 debug: getenv

2021/11/11 10:32:50 getenv duration: 13.9µs, args: AWS_DEFAULT_REGION
2021/11/11 10:32:50 debug: getenv

2021/11/11 10:32:50 getenv duration: 13.8µs, args: AWS_CONFIG_FILE
2021/11/11 10:32:50 error: getFileContent
open /home/rok/.aws/config: no such file or directory
2021/11/11 10:32:50 getFileContent duration: 87.4µs, args: /home/rok/.aws/config
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 isRunningAsRoot duration: 2.6µs, args:
2021/11/11 10:32:50 getcwd duration: 1.3µs, args:
2021/11/11 10:32:50 executionTime duration: 1.5µs, args:
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 lastErrorCode duration: 1.4µs, args:
2021/11/11 10:32:50 lastErrorCode duration: 1.3µs, args:
2021/11/11 10:32:50 lastErrorCode duration: 1.3µs, args:
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 getShellName duration: 1.4µs, args:
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 error: doGet
Get "http://127.0.0.1:9863/query": dial tcp 127.0.0.1:9863: connect: connection refused
2021/11/11 10:32:50 doGet duration: 782.5µs, args: http://127.0.0.1:9863/query
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 error: getBatteryInfo
[{Current:open /sys/class/power_supply/battery/charge_now: no such file or directory Full:open /sys/class/power_supply/battery/charge_full: no such file or directory Design:open /sys/class/power_supply/battery/charge_full_design: no such file or directory}]
2021/11/11 10:32:50 getBatteryInfo duration: 407µs, args:
2021/11/11 10:32:50 getcwd duration: 1.4µs, args:
2021/11/11 10:32:50 #### end oh-my-posh run ####

**Reply by JanDeDobbeleer** [reply]
Aha! It seems this is the culprit:

```log
2021/11/11 10:32:50 error: runCommand
exit status 128
2021/11/11 10:32:50 runCommand duration: 425.7431ms, args: git.exe -C /mnt/c/code/somerepo/ --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --short --branch
```

Can you update to `6.2.3`? That should resolve the issue. The command `git.exe -C /mnt/c/code/somerepo/ --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --short --branch` doesn't work (probably due to a Windows Path on WSL2) but `git.exe --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --short --branch` will work. @lnu might be useful info for you as well, I fear we will still have this issue on WSL2 regardless.

**Reply by roklenardic** [reply]
Yay, this works now, awesome support Jan, kudos! Btw, I am using WSL2 on win10 and used scoop to install and update
